packages/protobuf/src/private/binary-format-common.ts

```diff
@@ -235,8 +235,38 @@ function readMapEntry(
 }
 
 function readScalar(reader: IBinaryReader, type: ScalarType): any {
-  let [, method] = scalarTypeInfo(type);
-  return reader[method]();
```

Comment:
It might be lower impact to use `reader[method].call(reader)`

Reply:
That doesn't work either. Here's a flame graph with the switch statement:
![image](https://user-images.githubusercontent.com/5054866/232638259-55b0fa3a-fa98-458e-899b-6b104d64a0ef.png)

And with .call():
![image](https://user-images.githubusercontent.com/5054866/232638307-9e5fb463-0128-492c-9ebf-08b7d5b1396e.png)

Note that in the top chart, `string()` is >90% of the time spent in `readScalar()`. In the lower chart, around 1/3 of the time is spent in V8 something-something (I wish I could get more detail on what's going on there, but it's out of my depth and Google hasn't been helpful). The benchmark also comes out significantly worse.

Reply:
Oh I suspect the performance issue is the use of `scalarTypeInfo()`, which requires allocating an array to hold the return values, and does `const method = ScalarType[type].toLowerCase()` to determine the method. ToLowerCase is probably consuming some additional time.

Reply:
It's not `scalarTypeInfo()`; you can see that function in the second flame graph:

![image](https://user-images.githubusercontent.com/5054866/232840430-77ab5af5-1218-45cd-b77c-fed13d1bfd89.png)

My PR description includes a link to my benchmarking code, so I'd encourage you to clone this PR and that one, and merge them together to test any theories you have. I'm using https://clinicjs.org/flame/ to make flame graphs and invoking the benchmark with `clinic flame -- node --no-turbo-inlining dist/esm/benchmark.js` in order to keep functions separate in the result.